packages/react/src/create-strict-api/types.ts

```diff
@@ -0,0 +1,63 @@
+import type {
+  StrictCSSProperties,
+  CSSPseudoClasses,
+  CSSPseudoElements,
+  CSSPseudos,
+} from '../types';
+
+/**
+ * This is the shape of the generic object that `createStrictAPI()` takes.
+ * It's deliberately a subset of `AllowedStyles` and does not take at rules
+ * and pseudo elements.
+ */
+export type CompiledSchemaShape = StrictCSSProperties & {
+  [Q in CSSPseudoClasses]?: StrictCSSProperties;
+};
+
+export type PseudosDeclarations = { [Q in CSSPseudos]?: StrictCSSProperties };
+
+export type AllowedStyles = StrictCSSProperties & PseudosDeclarations;
+
+export type ApplySchemaValue<
+  TSchema,
+  TKey extends keyof StrictCSSProperties,
+  TPseudoKey extends CSSPseudoClasses | ''
+> = TKey extends keyof TSchema
+  ? // TKey is a valid property on the schema
+    TPseudoKey extends keyof TSchema
+    ? TKey extends keyof TSchema[TPseudoKey]
+      ? // We found a more specific value under TPseudoKey.
+        TSchema[TPseudoKey][TKey]
+      : // Did not found anything specific, use the top level TSchema value.
+        TSchema[TKey]
+    : // Did not found anything specific, use the top level TSchema value.
+      TSchema[TKey]
+  : // TKey wasn't found on the schema, fallback to the CSS property value
+    StrictCSSProperties[TKey];
```

Comment:
@itsdouges On second thought (from DMs), we could feasibly even `never` this until we're confident in our defensive API…

Reply:
Yeah, more of a question of how much we're worried about the unknown.  Either way, I gather we just make a 100% strict `CSS.Properties`-like interface ourselves to be honest, so if our `keyof TSchema` === `keyof StrictCSSProperties` then this doesn't matter.